app/frontend/src/features/profile/view/leaveReference/formSteps/Text.tsx

```diff
@@ -0,0 +1,88 @@
+import { useMediaQuery, useTheme } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import TextBody from "components/TextBody";
+import TextField from "components/TextField";
+import {
+  NEXT,
+  PUBLIC_ANSWER,
+  SUBMIT_STEP,
+  TEXT_EXPLANATION,
+} from "features/profile/constants";
+import ReferenceStepHeader from "features/profile/view/leaveReference/formSteps/ReferenceStepHeader";
+import {
+  ReferenceContextFormData,
+  ReferenceStepProps,
+  useReferenceStyles,
+} from "features/profile/view/leaveReference/ReferenceForm";
+import { ReferenceType } from "pb/references_pb";
+import { Controller, useForm } from "react-hook-form";
+import { useHistory, useParams } from "react-router-dom";
+import { leaveReferenceBaseRoute, referenceTypeRoute } from "routes";
+
+export default function Text({
+  user,
+  referenceData,
+  setReferenceValues,
+}: ReferenceStepProps) {
+  const history = useHistory();
+  const classes = useReferenceStyles();
+  const theme = useTheme();
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { referenceType, hostRequestId } =
+    useParams<{
+      referenceType: string;
+      hostRequestId?: string;
+    }>();
+  const { control, handleSubmit, errors } = useForm<ReferenceContextFormData>({
+    defaultValues: {
+      text: referenceData.text,
+    },
+  });
+
+  const onSubmit = handleSubmit((values) => {
+    setReferenceValues(values);
+    referenceType === referenceTypeRoute[ReferenceType.REFERENCE_TYPE_FRIEND]
+      ? history.push(
+          `${leaveReferenceBaseRoute}/${referenceType}/${user.userId}/${SUBMIT_STEP}`
+        )
+      : history.push(
+          `${leaveReferenceBaseRoute}/${referenceType}/${user.userId}/${hostRequestId}/${SUBMIT_STEP}`
+        );
+  });
+
+  return (
+    <form className={classes.form} onSubmit={onSubmit}>
+      <ReferenceStepHeader name={user.name} referenceType={referenceType} />
+      <TextBody className={classes.text}>{TEXT_EXPLANATION}</TextBody>
+      <TextBody className={classes.text}>{PUBLIC_ANSWER}</TextBody>
+      {errors && errors.text?.message && (
+        <Alert className={classes.alert} severity="error">
+          {errors.text.message}
+        </Alert>
+      )}
+      <div className={classes.card}>
+        <Controller
+          render={({ onChange }) => (
+            <TextField
+              className="multiline"
+              fullWidth={true}
+              multiline={true}
+              rows={15}
+              id="reference-text-input"
+              onChange={onChange}
+            />
+          )}
+          name="text"
+          control={control}
+          class={classes.card}
+        />
```

Comment:
This step needs some basic validation to make sure the user has at least typed something... I only spotted this as I noticed on the testing preview site it's possible for users to leave blank references, which isn't great.

Reply:
Silly oversight on my end, that's my bad.